infra/gcp/modules/gke-cluster/main.tf

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Create GCP SA for nodes
+resource "google_service_account" "cluster_node_sa" {
```

Comment:
You should consider adding a random string to SA's. The reason for this is if there is a change to an SA, its deleted and remade for example the Workload Identity mapping will break and be cached and tied to the deleted name. If you use a random string you can guarantee the mapping will be accurate. 
example
https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/sa.tf#L26C25-L26C25